libraries/ms-common/src/ms_test_helpers.c

```diff
@@ -0,0 +1,24 @@
+#include "ms_test_helpers.h"
+#include <string.h>
+#include <time.h>
+#include "log.h"
+#include "ms_test_helper_can.h"
+#include "status.h"
+
+#define CLOCK_HEADSTART 30000
+
+StatusCode time_assert(uint8_t value_to_compare, uint8_t *value_to_test, double time_ms,
+                       double tolerance_ms) {
+  double elapsed = 0;
+  double upper_bound = (double)(time_ms + tolerance_ms) / 1000.0;
+  double lower_bound = (double)(time_ms - tolerance_ms) / 1000.0;
+  clock_t begin = clock() - CLOCK_HEADSTART;

```

Comment:
What's `CLOCK_HEADSTART`, is that the approx number of clock ticks that it takes to start up? Seems like something that should be part of the tolerance I think. Also perhaps the tolerance should be a constant so we don't have to specify it everywhere? I can't see it really changing too much between tests.